app/client/src/workers/validations.ts

```diff
@@ -610,6 +643,13 @@ export const VALIDATORS: Record<ValidationTypes, Validator> = {
       return invalidResponse;
     }
     if (isString(value)) {
+      if (value === "" && !config.params?.required) {
+        return {
+          isValid: true,
+          parsed: config.params?.default || moment().toISOString(true),
```

Comment:
```suggestion
          parsed: config.params?.default || "",
```
Should it be an empty string instead? So that empty date inputs are supported and we can check if the binding value is set or it's in pristine state

Reply:
Actually this is existing behaviour on prod so if the field is empty referring to this field elsewhere(say for e.g {{Datepicker1.defaultData}} in say a label of Button widget) it would have the value of the current date. 
I thought changing this might deviate from existing behaviour for users, so I think it is better to do this change as a separate issue. 
Here the only intention was to not show it as an error.

Reply:
https://user-images.githubusercontent.com/67054171/131784920-a11881a0-f958-41c2-8216-4f974c370e9a.mov

Reply:
Yeah it looks like it sets to an empty string, but the actual value isn't an empty string.